src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -253,21 +273,61 @@ class OidcAuthorizeService implements AuthorizeService {
 export class AuthenticationService {
 
     static _infrastructureKey = 'Microsoft.AspNetCore.Components.WebAssembly.Authentication';
-    static _initialized : Promise<void>;
+    static _initialized: Promise<void>;
     static instance: OidcAuthorizeService;
+    static _pendingOperations: { [key: string]: Promise<AuthenticationResult> | undefined } = {}
 
     public static async init(settings: UserManagerSettings & AuthorizeServiceSettings) {
         // Multiple initializations can start concurrently and we want to avoid that.
         // In order to do so, we create an initialization promise and the first call to init
         // tries to initialize the app and sets up a promise other calls can await on.
         if (!AuthenticationService._initialized) {
-            this._initialized = (async () => {
-                const userManager = await this.createUserManager(settings);
+            AuthenticationService._initialized = AuthenticationService.InitializeCore(settings);
+
+            await AuthenticationService._initialized;
+        }
+
+        return AuthenticationService._initialized;
+    }
+
+    public static handleCallback() {
+        return AuthenticationService.InitializeCore();
+    }
+
+    private static async InitializeCore(settings?: UserManagerSettings & AuthorizeServiceSettings) {
```

Comment:
```suggestion
    private static async initializeCore(settings?: UserManagerSettings & AuthorizeServiceSettings) {
```

(plus related updates)

Reply:
done